batavia/types/Str.js

```diff
@@ -233,7 +233,7 @@ String.prototype.__iadd__ = function(other) {
 };
 
 String.prototype.__isub__ = function(other) {
-    throw new batavia.builtins.NotImplementedError("String.__isub__ has not been implemented");
+    throw new batavia.builtins.TypeError("unsupported operand type(s) for -=: 'str' and '" + batavia.type_name(other) +"'")
```

Comment:
Could you look at e916704fdb4a9c629b86bdb06324509ec3c099d2? I think the approach taken there is better, because it will be easier to change when we implement the "good" behavior.